lesson33/src/js/drawer-menu.js

```diff
@@ -45,3 +46,38 @@ const settingDrawerMenu = (options = {}) => {
 
 settingDrawerMenu();
 toggleButtonInDrawerMenu.addEventListener("click", toggleDrawerMenu);
+
+const fadeInAnimation = (element, duration) => {
+  element.animate([{ opacity: 0 }, { opacity: 1 }], duration);
+};
+
+fadeInAnimation(body, 300);
+body.classList.remove("fade-in");
+
+const transformLoadingAnimation = (element, duration) => {
+  return element.animate(
+    [
+      { transform: "translate(0,-50%)" },
+      { transform: "translate(-100%,-50%)" }
+    ],
+    duration
+  );
+};
+
+const transitionPageAnimation = (href) => {
+  body.classList.add("fade-in", "transition-animation");
+  const loading = document.getElementById("js-transition-loading");
+  const loadingAnimation = transformLoadingAnimation(loading, 500);
+  loadingAnimation.addEventListener(
+    "finish",
+    () => (window.location.href = href)
+  );
```

Comment:
I learned for the first time that an animation object is created by Element.animate() and that the object's animation can be monitored with a "finish" event.
With this, it would be possible to add some class and style the animation after it completes.
I learned a lot! Thank you!

Reply:
Thanks for the review. I'm glad I could be of some help:)